crowbar_framework/app/models/api/upgrade.rb

```diff
@@ -278,13 +278,15 @@ def upgrade_controller_nodes
         return false unless non_founder_api.disable_pre_upgrade_attribute_for founder.name
 
         # 3. delete old pacemaker resources
-        # FIXME: couldn't we do it right after migrating l3 routers from the first node?
         delete_pacemaker_resources non_founder.name
 
-        # 4. start crowbar-join at the first node
+        # 4. migrate routers from nodes being upgraded
+        return false unless founder_api.router_migration
```

Comment:
After looking at the code, I think we do not deal with a separated network cluster properly with our present code. We are basing the controller upgrade on the notion of [pacemaker] founder node and we assume it is also a network node (we move routers away from it).

The change you propose is right, but it involves more steps in order to make the solution fully compatible with a separated network cluster. I'm proposing at this point we still consider all clusters are in the same nodes and later on handle the network cluster separation. What do you think?